cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -97,7 +97,7 @@ void VhpiImpl::get_sim_precision(int32_t *precision)
     /* The value returned is in number of femtoseconds */
     vhpiPhysT prec = vhpi_get_phys(vhpiResolutionLimitP, NULL);
     uint64_t femtoseconds = ((uint64_t)prec.high << 32) | prec.low;
-    double base = 1e-15 * femtoseconds;
+    double base = 1e-15 * static_cast<double>(femtoseconds);
     *precision = (int32_t)log10(base);
```

Comment:
The compiler is rightly warning here - can we add a TODO, perhaps:
```
/* TODO: compute log10(base) as an integer, without losing precision by converting to double 
 * (or quantify whether this matters) */
```
Of course, if all the simulators do it this way anyway, then ignore this comment

Reply:
a flooring `log10` on an integer is easy enough to implement. All the other interfaces just ask the simulator "what's your precision"? But apparently VHPI deprecated that feature (???).

Reply:
Seems like a weird feature to me anyway, why restrict precision to powers of 20?